import/index_java/src/org/vufind/index/FormatCalculator.java

```diff
@@ -261,26 +283,67 @@ protected String getFormatFromBibLevel(Record record, char bibLevel, char format
      *
      * @param Record record
      * @param char recordType
+     * @param ControlField marc008
+     * @param List formatCodes007
      * @return String
      */
-    protected String getFormatFromRecordType(Record record, char recordType) {
+    protected String getFormatFromRecordType(Record record, char recordType, ControlField marc008, List formatCodes007) {
         switch (recordType) {
             case 'c':
             case 'd':
                 return "MusicalScore";
             case 'e':
             case 'f':
+                // Check 008/25 Type of cartographic material
+                switch (get008Value(marc008, 25)) {
+                    case 'd':
+                        return "Globe";
+                    case 'e':
+                        return "Atlas";
+                    default: break;
+                }
                 return "Map";
             case 'g':
-                // We're going to rely on the 007 instead for Projected Media
-                //return "Slide";
-                return "";
+                // Check 008/33 Type of visual material
+                switch (get008Value(marc008, 33)) {
+                    case 'f':
+                        return "Filmstrip";
+                    case 't':
+                        return "Transparency";
+                    case 'm':
+                        return "MotionPicture";
+                    case 'v': // Videorecording
+                        return "Video";
+                    default: break;
+                }
+                // Insufficient info in LDR and 008 to distinguish still from moving images
```

Comment:
@demiankatz we could probably try one more thing here:
-- if 008/34 is set to something, it should mean it is a video: "Technique - One-character alphabetic code that indicates the technique used in creating motion in motion pictures or videorecordings."

In your records, it is is set to 'c' ("Animation and live action") even though 008/33 Type of visual material is 'z' ("other"). So on this basis we could set default to "Video" instead of "Slide". What do you think?

Reply:
That sounds reasonable to me!

Reply:
If you make this last change, I can run one more reindex and see if that cleans up the remaining "Projected Medium" values.

Reply:
done